src/main/java/net/neoforged/neoforge/registries/DeferredRegister.java

```diff
@@ -648,6 +663,29 @@ public <D> DeferredHolder<DataComponentType<?>, DataComponentType<D>> registerCo
         }
     }
 
+    /**
+     * Specialized DeferredRegister for {@link EntityType EntityTypes}.
+     */
+    public static class EntityTypes extends DeferredRegister<EntityType<?>> {
+        protected EntityTypes(String namespace) {
+            super(Registries.ENTITY_TYPE, namespace);
+        }
+
+        /**
+         * Convenience method that constructs a builder for use in the operator. Use this to avoid inference issues.
+         *
+         * @param name     The name for this entity type. It will automatically have the {@linkplain #getNamespace() namespace} prefixed.
+         * @param factory  The factory used to typically construct the entity when using an existing helper from the type.
+         * @param category The category of the entity, typically {@link MobCategory#MISC} for non-living entities, or one of the others for living entities.
+         * @param builder  The unary operator, which is passed a new builder for user operators, then builds it upon registration.
+         * @return A {@link DeferredHolder} which reflects the data that will be registered.
+         * @param <E> the type of the entity
+         */
+        public <E extends Entity> DeferredHolder<EntityType<?>, EntityType<E>> registerType(String name, EntityType.EntityFactory<E> factory, MobCategory category, UnaryOperator<EntityType.Builder<E>> builder) {
```

Comment:
I wonder if it should be called `registerEntityType` to match `registerComponentType` above.

Reply:
I have no opinion on this, so I can rename if desired.